pyshelf/search/update_manager.py

```diff
@@ -1,9 +1,46 @@
 from pyshelf.search.metadata import Metadata
+from elasticsearch_dsl.query import Q
+from elasticsearch_dsl import Search
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import scan, bulk
 
 
 class UpdateManager(object):
-    def __init__(self, logger):
+    def __init__(self, logger, host, index):
         self.logger = logger
+        self.connection = Elasticsearch(host)
+        self.index = index
+
+    def remove_unlisted_documents(self, ex_key_list):
+        """
+            Removes any documents with keys not enumerated in the ex_key_list.
+
+            Args:
+                ex_key_list(list): List of keys that should remain in the Elasticsearch collection.
+
+            Returns:
+                Number of documents deleted from Elasticsearch.
+        """
+        query = ~Q("ids", type=Metadata._doc_type.name, values=ex_key_list)
```

Comment:
Could you make a note of whats going on here.  I assume the invert operator makes it like the opposite of that query or something